### PR TITLE
Adding Nginx routing SPI for HA applications

### PR DESCRIPTION
**Pull request by andrewklau**

### CODE REVIEW

**Comment by sdodson** [comment]
[test] test results were purged

**Comment by sdodson** [comment]
@andrewklau Can you view this? I don't think you have to be on Red Hat's VPN for it but I'm not 100% certain. If so search for ') Error:' to find the test failures.

https://ci.openshift.redhat.com/jenkins/job/test_pull_requests/3231/consoleFull

The tests are failing because they expect ALLOW_HA_APPLICATIONS="true". We could only adjust DEFAULT_ALLOW_HA, leaving ALLOW_HA_APPLICATIONS="true" regardless of whether routing is enabled or not. While that seems like a hack to work around these tests, it is how the default config is deployed.  However I don't think there's much point in ALLOW_HA_APPLICATIONS='true' without an HA routing layer. So even ignoring the automated test suite for a minute, I'm not sure what the best course of action here is.

Specific to the test suite, I can modify it to enable routing.

**Comment by andrewklau** [comment]
@sdodson Thanks, yes I can. 
Let me know if I need to modify anything else here, perhaps I could modify it for the test suite and return it back if it's successful.

File: manifests/init.pp [hunk 1]
```diff
@@ -1041,4 +1067,8 @@
     Class['openshift_origin::update_conf_files'] ->
     class { 'openshift_origin::role::load_balancer': }
   }
+  if member( $roles, 'nginx' ) {
```
Comment (Filirom1) [inline manifests/init.pp]: role is `routing` not `nginx`

File: manifests/init.pp [hunk 2]
```diff
@@ -1041,4 +1067,8 @@
...
+    Class['openshift_origin::update_conf_files'] ->
+    class { 'openshift_origin::role::nginx': }
```
Comment (Filirom1) [inline manifests/init.pp]: role is `routing` not `nginx`

**Pull request closed without merging**